src/Raven.Studio/typescript/viewmodels/manage/editServerWideBackup.ts

```diff
@@ -0,0 +1,251 @@
+    
+    activate(args: any) {
+        super.activate(args);
+
+        const backupLoader = () => {
+            const deferred = $.Deferred<void>();
+
+            if (args && args.taskName) { 
+                // 1 Editing an existing task
+                this.isAddingNewBackupTask(false);
+               
+                new getServerWideBackupsCommand(args.taskName)
+                    .execute()
+                    .done((backupsList: Raven.Server.Web.System.ServerWideBackupConfigurationResults) => {
+                        if (backupsList.Results.length) {
+                            const backupTask = backupsList.Results[0];
+                            if (this.serverConfiguration().LocalRootPath && backupTask.LocalSettings.FolderPath && backupTask.LocalSettings.FolderPath.startsWith(this.serverConfiguration().LocalRootPath)) {
+                                backupTask.LocalSettings.FolderPath = backupTask.LocalSettings.FolderPath.substr(this.serverConfiguration().LocalRootPath.length);
+                            }
+
+                            this.configuration(new periodicBackupConfiguration(backupTask, this.serverConfiguration(), false, true)); 
+                            deferred.resolve();
+                        }
+                        else {
+                            deferred.reject();
+                            router.navigate(appUrl.forServerWideBackupList());
+                        }
+                    })
+                   .fail(() => {
+                         deferred.reject();
+                         router.navigate(appUrl.forServerWideBackupList()); 
+                    });
+            
+            } else {
+                // 2. Creating a new task
+                this.isAddingNewBackupTask(true);
+                
+                this.configuration(periodicBackupConfiguration.empty(this.serverConfiguration(), false, true));
+                deferred.resolve();
+            }
+
+            return deferred
+                .then(() => {
+                    const encryptionSettings = this.configuration().encryptionSettings();
+                    this.encryptionSection = setupEncryptionKey.forServerWideBackup(encryptionSettings.key, encryptionSettings.keyConfirmation); 
+                    
+                    this.dirtyFlag = this.configuration().dirtyFlag;
+
+                    this.fullBackupCronEditor(new cronEditor(this.configuration().fullBackupFrequency));
+                    this.incrementalBackupCronEditor(new cronEditor(this.configuration().incrementalBackupFrequency));
+
+                    if (!encryptionSettings.key()) {
+                        return this.encryptionSection.generateEncryptionKey()
+                            .done(() => {
+                                encryptionSettings.dirtyFlag().reset();
+                            });
+                    }
+                });
+        };
+
+        return $.when<any>(this.loadServerSideConfiguration())
```

Comment:
no need to $.when here - you have single task to await.

Reply:
done